lib/hammer_cli_foreman_discovery/discovery.rb

```diff
@@ -129,6 +129,19 @@ class AutoProvisionCommand < HammerCLIForeman::SingleResourceCommand
       action :auto_provision
       command_name "auto-provision"
       desc _("Auto provision a host")
+
+      option '--all', :flag, _("Auto provision all discovered hosts")
+
+      def execute
+        if option_all?
+          resource.call(:auto_provision_all, {})
+          print_message _("Hosts created") if success_message
```

Comment:
I'm curious if this prints anything. The `success_message` usually defines the message that should be printed on success (not if it should be printed or not). Here it is not defined so the condition is going to be false always. 

IMO the condition should be removed as the resource.call raises exception for non-200 responses (and the `print_massage` is skipped) thus the L138 is reached only on success.

From hammer perspective it looks reasonable otherwise. 
Note: I didn't tested this PR.

Reply:
Yes, tested it. There is no need for the `if success_message` as it just prints the message for success. 
Thanks @mbacovsky for the input :)

![screenshot from 2017-11-14 12-09-53](https://user-images.githubusercontent.com/10976840/32766398-b8bbdeb2-c906-11e7-8c8c-0f67ec2961e2.png)